Xamarin.Forms.Platform.Android/CollectionView/SelectableItemsViewAdapter.cs

```diff
@@ -65,6 +56,69 @@ internal void ClearNativeSelection()
 			}
 		}
 
+		internal void MarkNativeSelection(object selectedItem)
+		{
+			if (selectedItem == null)
+			{
+				return;
+			}
+
+			var position = GetPositionForItem(selectedItem);
+
+			for (int i = 0; i < _currentViewHolders.Count; i++)
+			{
+				if (_currentViewHolders[i].AdapterPosition == position)
+				{
+					_currentViewHolders[i].IsSelected = true;
+					return;
+				}
+			}
+		}
+
+		int[] GetSelectedPositions()
+		{
+			switch (SelectableItemsView.SelectionMode)
+			{
+				case SelectionMode.None:
+					return new int[0];
+
+				case SelectionMode.Single:
+					var selectedItem = SelectableItemsView.SelectedItem;
+					if (selectedItem == null)
+					{
+						return new int[0];
+					}
+
+					return new int[1] { GetPositionForItem(selectedItem) };
+
+				case SelectionMode.Multiple:
+					var selectedItems = SelectableItemsView.SelectedItems;
+					var result = new int[selectedItems.Count];
+
+					for (int n = 0; n < result.Length; n++)
+					{
+						result[n] = GetPositionForItem(selectedItems[n]);
+					}
+
+					return result;
+			}
+
+			return new int[0];
+		}
+
+		bool PostionIsSelected(int position)
+		{
+			foreach (var selectedPosition in GetSelectedPositions())
```

Comment:
I think it’s better to add a variable instead of a `GetSelectedPositions` method in `foreach` to avoid a multiple call when iterating